test/e2e/navigation.test.js

```diff
@@ -1,17 +1,40 @@
 /* global it expect beforeAll afterAll */
 import ms from 'milliseconds'
 import { launch, appUrl } from './puppeteer'
+import IPFS from 'ipfs'
 
 let browser
+let node
 
-beforeAll(async () => { browser = await launch() })
-afterAll(() => browser.close())
+beforeAll(done => {
+  node = new IPFS()
 
-it('Navigation test', async () => {
+  node.on('ready', async () => {
+    browser = await launch()
+    done()
+  })
+})
+
+afterAll(async () => {
+  browser.close()
+  await node.stop()
+})
+
+it('Navigation test: node not running', async () => {
+  const page = (await browser.pages())[0]
+  await page.goto(appUrl)
+  page.waitForFunction(`document.title === 'Welcome to IPFS'`, {timeout: 5000})
+}, ms.minutes(1))
+
+it('Navigation test: node running', async () => {
   // Save 3s per run by using the initial page rather than creating a new one!
   // const page = await browser.newPage()
   const page = (await browser.pages())[0]
 
+  await page.evaluate(() => {
+    window.ipfs = node
```

Comment:
@hacdias ok so, this is a rad idea, but I don't think it'll work. With the e2e tests things get complicated, as the test is running in jest in node, and firing up a browser instance with it's own seperate runtime. I don't think we'll ever be able to create an ipfs instance in a node runtime and smoosh it into a browser without very weird things happening.

Much more minor, the api for `page.evaluate` requires you to pass any args you want to hand to the page, and they get magically transferred as args to the callback function when it is executed in the context of the page instance. Even so, like going through a black hole, an ipfs instance won't survive the trip.

One solution is to create a fake window.ipfs object with an id function that resolves with a test identity info, and then check for that. We don't need to test that js-ipfs works in the webui tests, so it's ok to stub it out.